main.py

```diff
@@ -1112,6 +1112,18 @@ def patch_record_in_place(fn, record, subdir):
     if subdir.startswith("win-"):
         replace_dep(depends, "zeromq >=4.3.1,<4.4.0a0", "zeromq >=4.3.1,<4.3.2.0a0")
 
+    # Up to libtiff 4.4.0, the LIBTIFF_CURRENT version was 13, and LIBTIFF_AGE was 8.
+    # As of 4.5.0, LIBTIFF_CURRENT was decremented to 6, and LIBTIFF_AGE was reset.
+    # This change has major implications for Mach-O dynamic libraries on MacOS.
+    # We applied set_configure_script_version.patch in v4.5.0 to fix the issue.
+    # In v4.7.0 we removed that patch, so we have to constrain libtiff at runtime to <4.7.0.
+    if subdir.startswith("osx-"):
+        replace_dep(depends, "libtiff >=4.1.10,<5.0a0", "libtiff >=4.1.10,<4.7.0")
```

Comment:
```suggestion
        replace_dep(depends, "libtiff >=4.0.8,<5.0a0", "libtiff >=4.0.8,<4.7.0")
        replace_dep(depends, "libtiff >=4.0.9,<5.0a0", "libtiff >=4.0.9,<4.7.0")
        replace_dep(depends, "libtiff >=4.0.10,<5.0a0", "libtiff >=4.0.10,<4.7.0")
```
I ran `ddb query get-package-artifacts --depends-on="^libtiff($| .*)" | jq > out.txt`
Then filtered on libtiff, and sorted with unique entries.